### ISSUE TITLE
Seriously get rid of TIMER_QUANTUM and burn the memory.

### ISSUE THREAD

**Issue by digitalextremist** [celluloid/culture#24]
0. We ought to be using a monotonic timer, if using a timer at all.
0. The variance between platforms, and the influence of CI load makes the number untenable.
0. :fire:

**Comment by tarcieri** [comment]
The timers gem uses the hitimes gem which provides portable monotonic
timers although I'm not sure that all usages of non-monotonic time have
been expunged from the entire codebase.

The timer quantums in the tests can probably be replaced by basic event
ordering guarantees that shouldn't be violated without hardcoding magic
time intervals.

On Thursday, August 6, 2015, Donovan Keme <notifications@github.com> wrote:

>
>    1. We ought to be using a monotonic timer, if using a timer at all.
>    2. The variance between platforms, and the influence of CI load makes
>    the number untenable.
>    3. [image: :fire:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/celluloid/culture/issues/24>.
>


-- 
Tony Arcieri

**Comment by digitalextremist** [comment]
Agreed.. I was expecting `timers` ( `hitimes` ) ... that'd be perfect. I will do a monotonic sweep. I think I'm going to do #2 and then do that and a few other things. Before that though, I need to push a `timers` point release. When you get a chance can you give me cut rights on rubygems for that? I have everything but that I think. And IRC ops so I can change the topic as versions advance. Don't bother with this at DEFCON though. Just filing it with you.
